src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs

```diff
@@ -176,44 +224,65 @@ protected virtual void OnTextChanged(DependencyPropertyChangedEventArgs e)
 #endif
 				{
 					_isInvokingTextChanged = false;
+					_isTextChangedPending = false;
 				}
 			}
 
+			_textBoxView?.SetTextNative(Text);
+		}
+
+		private void UpdatePlaceholderVisibility()
+		{
 			if (_placeHolder != null)
 			{
 				_placeHolder.Visibility = Text.IsNullOrEmpty() ? Visibility.Visible : Visibility.Collapsed;
 			}
+		}
 
-			UpdateButtonStates();
+		private object CoerceText(object baseValue)
+		{
+			if (!(baseValue is string baseString))
+			{
+				return DependencyProperty.UnsetValue; //TODO: UWP throws ArgumentNullException, in principle we should do the same. 
```

Comment:
How can we validate that ?

Reply:
I already added a unit test for it: https://github.com/unoplatform/uno/pull/1215/files#diff-9fe81dcf3075e44b3ff13420a735cd47R44 If we ever adjust this behaviour we'd remove the `#if NETFX_CORE`.

(We could change it right now to throw an exception, I just didn't do it because it seems like a breaking change with little gain. Eg, if Uno's binding engine pushes a null in some cases where UWP doesn't.)